languages/elixir/exercises/concept/basic-strings/.meta/design.md

```diff
@@ -0,0 +1,51 @@
+# Design
+
+## Goal
+
+The goal of this exercise is to teach the student how strings are implemented and used in Elixir.
+
+## Learning objectives
+
+After completing this exercise, the student should:
+
+- Know how to concatenate strings.
+- Know how to manipulate strings.
+- Know how to perform string interpolation.
+- Know how to create multiline strings (heredocs).
+
+## Out of scope
+
+- Strings as binaries for binary pattern matching.
+- Regular expressions.
+- Sigils.
+- Charlists.
+
+## Prerequisites
+
+This exercise's prerequisites are:
+
+- `lists`: needs to be able to work with the return value of `String.split`.
+- `basic-pattern-matching`: needs to be able to work with the return value of `String.split`.
```

Comment:
Technically pattern-matching is not necessary to deal with a two-element list. Based on the `lists` concept exercise, somebody could write:
```elixir
names = String.split("John Doe")
first_name = hd(names)
last_name = tl(hd(names))
```

But that doesn't look great. This is much nicer:
```elixir
[first_name, last_name] = String.split("John Doe")
```

I allowed myself to `basic-pattern-matching` here because it was listed on the issue, but I don't see any exercises yet that teach it.

Reply:
I have a PR for `tuples` which opens the `basic-pattern-matching` pre-req, so I think it works good here to use it.  At this point, unless an analyzer is built for static code analysis, there is no way for us to police which form a student would use, but if they were to ask on elixirforum "how do I get each name from string `James Dean`, I believe they would be given an answer using pattern matching.

Reply:
> I allowed myself to `basic-pattern-matching` here because it was listed on the issue, but I don't see any exercises yet that teach it.

This is perfectly reasonable. We'll be able to figure out which Concepts don't have an exercise that unlocks them easily. It is totally fine to just assume that there will be an exercise at some point that unlocks a Concept.